crates/python/src/executable.rs

```diff
@@ -0,0 +1,162 @@
+use std::sync::Arc;
+
+use pyo3::{
+    exceptions::PyRuntimeError, pyclass, pymethods, types::PyDict, FromPyObject, Py, PyAny,
+    PyResult, Python,
+};
+use qcs::{Error, Executable, JobHandle, Qpu, Qvm, Service};
+use rigetti_pyo3::{
+    impl_as_mut_for_wrapper, py_wrap_error, py_wrap_simple_enum, py_wrap_type, wrap_error,
+    PyWrapper, ToPython, ToPythonError,
+};
+use tokio::sync::Mutex;
+
+use crate::qpu::quilc::PyCompilerOpts;
+
+wrap_error!(ExecutionError(Error));
+
+py_wrap_error!(
+    executable,
+    ExecutionError,
+    QcsExecutionError,
+    PyRuntimeError
+);
+
+// Because Python is garbage-collected, no lifetimes can be guaranteed except `'static`.
+//
+// `Arc<Mutex<>>` to work around https://github.com/awestlake87/pyo3-asyncio/issues/50
+py_wrap_type! {
+    PyExecutable(Arc<Mutex<Executable<'static, 'static>>>) as "Executable";
+}
+
+impl_as_mut_for_wrapper!(PyExecutable);
+
+#[pyclass]
+#[pyo3(name = "ExeParameter")]
+#[derive(FromPyObject)]
+pub struct PyParameter {
+    name: String,
+    index: usize,
+    value: f64,
+}
+
+#[pymethods]
+impl PyExecutable {
+    #[new]
+    #[args(
+        "/",
+        registers = "Vec::new()",
+        parameters = "Vec::new()",
+        shots = "None",
+        compile_with_quilc = "None",
+        compiler_options = "None"
+    )]
```

Comment:
This does two things:

1) Defines all parameters after the "/" as keyword-only
2) Sets defaults for all of the listed parameters (non-listed are required)

Reply:
Will require at least python 3.8 - so I think this needs to be updated https://github.com/rigetti/qcs-sdk-rust/blob/2722a59c9f2b472d5d607458be0d310e5d0a503d/crates/python/pyproject.toml#L43

Reply:
I'm curious if the `pyo3` stuff uses that PEP or if it does something else. I'm okay with bumping the Python version though, if @kalzoo does not see any issues.

Reply:
I hope so, since support EOL is in 5 months 😬

Reply:
I'm good with 3.8.